lib/sunbeam-ws.js

```diff
@@ -464,6 +485,121 @@ class MandelbrotEosfinex extends EventEmitter {
     return { payload }
   }
 
+  async deposit (payload) {
+    const { exchangeContract, expireInSeconds } = this.conf.eos
+    if (!Array.isArray(payload)) {
+      payload = [payload]
+    }
+
+    const [listedTokens, auth] = await Promise.all([
+      this.getListedTokens(),
+      this.getAuth()
+    ])
+
+    const txData = []
+    const actions = payload.map(txDetails => {
+      let { currency, amount, to = exchangeContract, memo = '' } = txDetails
+      if (!currency || !amount) {
+        throw new Error('Currency and amount should be set')
+      }
+      if (amount[0] === '-') {
+        throw new Error('Amount should be positive')
+      }
+
+      currency = currency.toUpperCase()
+      const tokenInfo = listedTokens[currency]
+      if (!tokenInfo) {
+        throw new Error('Currency is not supported: ' + currency)
+      }
+
+      txData.push({ currency, amount, to })
+
+      return {
+        account: tokenInfo.issuer,
+        name: 'transfer',
+        authorization: [{
+          actor: auth.account,
+          permission: auth.permission
+        }],
+        data: {
+          to,
+          from: auth.account,
+          quantity: `${decimalPad(amount, tokenInfo.precision)} ${currency}`,
+          memo
+        }
+      }
+    })
+
+    const txRes = await this.client.api.transact({ actions }, {
+      blocksBehind: 3,
+      broadcast: true,
+      expireSeconds: expireInSeconds
```

Comment:
this is used three times in the codebase, maybe we can put it in a central location?